source/extensions/common/wasm/context.cc

```diff
@@ -1125,13 +1132,30 @@ WasmResult Context::setProperty(absl::string_view path, absl::string_view value)
   if (!stream_info) {
     return WasmResult::NotFound;
   }
+  
   std::string key;
+  auto part_token = property_tokens.find(path);
```

Comment:
I think only metadata is mutable, so this seems unnecessary. Maybe just do a prefix test for `metadata\0` in path?